### PR TITLE
Pytorch 0.4 Support

### PR DESCRIPTION
**Pull request by xingdi-eric-yuan**
Basically the changes are:

- 1. torch.cat --> torch.stack
- 2. torch.split uses new argument name split_size_or_sections
- 3. module._backend.Embedding.apply changes, so just use torch.nn.Embedding
- 4. since pytorch 0.31 it checks when updating parameters, model must be under training mode. Since we are not using dropout anyway, just removed the model.eval line during agent.update

### CODE REVIEW

**Comment by anantzoid** [comment]
@xingdi-eric-yuan  It'll be great if you could also replace `orthogonal` and `xavier_uniform` to `orthogonal_` and `xavier_uniform_` respectively since the former are now deprecated.

**Comment by xingdi-eric-yuan** [comment]
@anantzoid nice catch, done.
